store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -123,6 +125,43 @@ public CarbonWriterBuilder uniqueIdentifier(long UUID) {
     return this;
   }
 
+  /**
+   * To support the load options for sdk writer
+   * @param options key,value pair of load options.
+   *                supported keys values are
+   *                a. bad_records_logger_enable -- true, false
+   *                b. bad_records_action -- FAIL, FORCE, IGNORE, REDIRECT
+   *                c. bad_record_path -- path
+   *                d. dateformat -- same as JAVA SimpleDateFormat
+   *                e. timestampformat -- same as JAVA SimpleDateFormat
+   * @return updated CarbonWriterBuilder
+   */
+  public CarbonWriterBuilder withLoadOptions(Map<String, String> options) {
```

Comment:
complex type level delimiters should be supported, quote char, escape character needs to be supported, as required for complex types parsing.

Reply:
ok. done